api/controllers/HomeController.js

```diff
@@ -31,18 +31,23 @@ module.exports = {
     });
   },
 
-  reference: function(req, res) {
-    User.findOne({name: req.params.user}).exec(function (err, user){
-      if (user) {
-        res.view();
-      } else {
-        res.view('404', {data: {user: req.params.user, error: "User not found"}});
+  reference: async function(req, res) {
+    try {
+      return res.view({refUser: await Users.get(req.user, req.params.user)});
```

Comment:
I'm a little bit confused as to the Users.get though, why are we sending it a user and a username? That's weird.